models/Asset/Image/Thumbnail.php

```diff
@@ -381,7 +381,7 @@ public function getHtml($options = [], $removeAttributes = [])
 
                     $sourceHtml = '<source ' . array_to_html_attribute_string($sourceTagAttributes) . ' />';
                     if ($isAutoFormat) {
-                        $sourceHtmlWebP = preg_replace(['@(\.)(pjpeg|png)@', '@(/)(jpeg|png)@'], '$1webp', $sourceHtml);
+                        $sourceHtmlWebP = preg_replace(['@(\.)(jpg|png)@', '@(/)(jpeg|png)@'], '$1webp', $sourceHtml);
```

Comment:
This looks like it is a little bit too greedy, if the substring ".png" or ".jpg" happens to appear inside the file name it might replace the incorrect part of the string:

```php
$sourceHtml = "<source src='test.pngenerated.jpg' />";

$result = preg_replace(['@(\.)(jpg|png)@', '@(/)(jpeg|png)@'], '$1webp', $sourceHtml);

print_r($result); // <source src='test.webpenerated.webp' />
```

Reply:
Yep makes sense to harden that, thanks for your feedback!